serviceApiKey added

sanket-shivam-vs opened the pull request.

sanket-shivam-vs committed:
serviceApiKey added

sanket-shivam-vs committed:
minor changes

siddharth-vs commented on src/services/apiMonitor.service.ts:
```diff
@@ -6,19 +6,31 @@ import { requestLogQueue } from '.';
       logger.init(logLevel);
 
-      await MongooseClient.init(mongoUrl);
+      if(process.env.mongoUrl) {
```
place the mongo url in constants instead of env

siddharth-vs commented on src/index.ts:
```diff
@@ -28,24 +31,25 @@ interface PluginOptions extends FastifyPluginOptions {
+
   const {
-    mongoUrl,
     organizationName,
```
plugin options need proper nesting
organizationName,mongoURL etc inside dbConfig etc....

siddharth-vs commented on src/services/userAccount.service.ts:
```diff
@@ -56,4 +56,18 @@ export class UserAccountService {
+  async setAccountInfo(
+    serviceKey: string,
+  ) {
+    const accountInfo:any = await this.microserviceDAO.getAccountDetailsByApiKey(serviceKey);
```
error handling missing
what if key is invalid

sanket-shivam-vs committed:
comments resolved

sanket-shivam-vs committed:
Get user account details from Api using api-key

Now if serviceKey is provided, service account details (projectId, orgId, microserviceId, mongoDbUrl of server) will be fetched from Saas Api -

getUserAccountDetails

sanket-shivam-vs committed:
changes

sanket-shivam-vs committed:
changes

sanket-shivam-vs committed:
Update package.json

siddharth-vs commented on src/services/apiMonitor.service.ts:
```diff
@@ -1,24 +1,56 @@
+      if (serviceKey) {
+
+        const userAccountInfo = await axiosClient.get(
+          'https://66a0bce27053166bcabc8e93.mockapi.io/api/v1/account/user',
```
place this in constants

sanket-shivam-vs committed:
moved axios url to constants
